datanode/sqlstore/cursor.go

```diff
@@ -218,6 +228,13 @@ func PaginateQuery[T any, PT parserPtr[T]](
 		query = fmt.Sprintf("%s %s (%s)", query, whereOrAnd, predicate)
 	}
 
+	groupBy := ""
+	if len(grouping) > 0 {
+		groupBy = fmt.Sprintf(" GROUP BY %s", strings.Join(grouping, ","))
+	}
+
+	query = fmt.Sprintf("%s%s", query, groupBy)
+
```

Comment:
I'm not sure we want to mix up grouping and paging, they're sort of separate operations.. Presumably we want to group a thing and then we've effectively got a new shape of table and we want to have a cursor into that possibly using a different cursor as the old one might not make sense if columns have disappeared?

Reply:
Unfortunately another by-product of the complex query for the aggregated balances where you need to have a group by clause in your query before the order by statement. Happy to discuss and re-implement this if there is a better way, but to me it makes sense as other queries in future may require a group by clause before the order by is constructed for the pagination.